packages/menu-surface/index.tsx

```diff
@@ -253,62 +250,60 @@ class MenuSurface extends React.Component<MenuSurfaceProps, MenuSurfaceState> {
         return {x: window.pageXOffset, y: window.pageYOffset};
       },
-      setPosition: (position: Position) => {
+      setPosition: (position: Partial<MDCMenuDistance>) => {
         this.setState((prevState) => Object.assign(prevState, {
           styleLeft: 'left' in position ? position.left : null,
           styleRight: 'right' in position ? position.right : null,
           styleTop: 'top' in position ? position.top : null,
           styleBottom: 'bottom' in position ? position.bottom : null,
         }));
       },
-      setMaxHeight: (maxHeight: number) => this.setState({maxHeight}),
+      setMaxHeight: (maxHeight: string) => this.setState({maxHeight}),
     };
 
-    return Object.assign(
-      {
-        addClass: (className: string) => {
-          const classList = new Set(this.state.classList);
-          classList.add(className);
-          this.setState({classList});
-        },
-        removeClass: (className: string) => {
-          const classList = new Set(this.state.classList);
-          classList.delete(className);
-          this.setState({classList});
-        },
-        hasClass: (className: string) => this.classes.split(' ').includes(className),
-        hasAnchor: () => !!this.props.anchorElement,
-        notifyOpen: () => {
-          if (this.registerWindowClickListener) {
-            this.registerWindowClickListener();
-          }
-          this.props.onOpen!();
-        },
-        notifyClose: () => {
-          if (this.deregisterWindowClickListener) {
-            this.deregisterWindowClickListener();
-          }
-          this.props.onClose!();
-        },
-        isElementInContainer: (el: HTMLElement) => {
-          if (!this.menuSurfaceElement.current) return false;
-          if (this.menuSurfaceElement.current === el) {
-            return true;
-          }
-          return this.menuSurfaceElement.current.contains(el);
-        },
-        isRtl: () => {
-          if (!this.menuSurfaceElement) return false;
-          if (!this.menuSurfaceElement.current) return false;
-          return window
-            .getComputedStyle(this.menuSurfaceElement.current)
-            .getPropertyValue('direction') === 'rtl';
-        },
-        setTransformOrigin: (transformOrigin: string) => this.setState({transformOrigin}),
+    return {
+      addClass: (className: string) => {
+        const classList = new Set(this.state.classList);
+        classList.add(className);
+        this.setState({classList});
       },
-      focusAdapterMethods,
-      dimensionAdapterMethods
-    );
+      removeClass: (className: string) => {
+        const classList = new Set(this.state.classList);
+        classList.delete(className);
+        this.setState({classList});
+      },
+      hasClass: (className: string) => this.classes.split(' ').includes(className),
+      hasAnchor: () => !!this.props.anchorElement,
+      notifyOpen: () => {
+        if (this.registerWindowClickListener) {
```

Comment:
Why do we need to register window listener here? Seems like this adapter doing more stuff than it should. We're not doing it in vanilla component implementation?

Reply:
It actually gets registered during the OPENED_EVENT `this.listen(strings.OPENED_EVENT, this.registerBodyClickListener_);`, which eventually calls the `#adapter.notifyOpen` method. Since we don't listen to the events, this is the closest we can get to the vanilla component. What are your thoughts?

Reply:
I feel it is clearer if we encapsulate these event registrations.

```
notifyOpen: () => {
  this.handleOpen_();
  this.props.onOpen!();
}
```

So you can put more stuff in `handleOpen_` in feature if need to be. Same for `notifyClose`.

WDYT?

Reply:
Ya that would clean up the adapter. Sounds good to me